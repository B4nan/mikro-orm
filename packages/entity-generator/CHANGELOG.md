# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [4.0.2](https://github.com/mikro-orm/mikro-orm/compare/v4.0.1...v4.0.2) (2020-09-11)


### Features

* **entity-generator:** do not use ts-morph ([478a7bb](https://github.com/mikro-orm/mikro-orm/commit/478a7bb7f9ea80062caaef666b8308086842a44b))





## [4.0.1](https://github.com/mikro-orm/mikro-orm/compare/v4.0.0...v4.0.1) (2020-09-10)

**Note:** Version bump only for package @mikro-orm/entity-generator
