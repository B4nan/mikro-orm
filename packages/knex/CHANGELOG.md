# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [4.0.3](https://github.com/mikro-orm/mikro-orm/compare/v4.0.2...v4.0.3) (2020-09-15)


### Performance Improvements

* **core:** do not use `em.merge()` internally ([6a1a6d6](https://github.com/mikro-orm/mikro-orm/commit/6a1a6d68b65a20b8f1a78bf644844427f3b2dd1a))





## [4.0.2](https://github.com/mikro-orm/mikro-orm/compare/v4.0.1...v4.0.2) (2020-09-11)

**Note:** Version bump only for package @mikro-orm/knex





## [4.0.1](https://github.com/mikro-orm/mikro-orm/compare/v4.0.0...v4.0.1) (2020-09-10)


### Bug Fixes

* **core:** refactor internals to reduce number of cycles ([#830](https://github.com/mikro-orm/mikro-orm/issues/830)) ([3994767](https://github.com/mikro-orm/mikro-orm/commit/3994767d93ef119d229bedffa77eb2ea3af5c775))
