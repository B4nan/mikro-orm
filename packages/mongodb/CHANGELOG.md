# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [4.2.3](https://github.com/mikro-orm/mikro-orm/compare/v4.2.2...v4.2.3) (2020-10-24)

**Note:** Version bump only for package @mikro-orm/mongodb





## [4.2.2](https://github.com/mikro-orm/mikro-orm/compare/v4.2.1...v4.2.2) (2020-10-22)

**Note:** Version bump only for package @mikro-orm/mongodb





## [4.2.1](https://github.com/mikro-orm/mikro-orm/compare/v4.2.0...v4.2.1) (2020-10-20)

**Note:** Version bump only for package @mikro-orm/mongodb





# [4.2.0](https://github.com/mikro-orm/mikro-orm/compare/v4.1.1...v4.2.0) (2020-10-18)


### Bug Fixes

* **core:** update umzug types to 2.3 ([4668e78](https://github.com/mikro-orm/mikro-orm/commit/4668e78d1c900d1718625364b603dda1b707dd82)), closes [#926](https://github.com/mikro-orm/mikro-orm/issues/926)


### Features

* **core:** allow defining multiple entities in single file ([e3ab336](https://github.com/mikro-orm/mikro-orm/commit/e3ab33699b116e25e09a7449589db7955899c8ac)), closes [#922](https://github.com/mikro-orm/mikro-orm/issues/922)
* **core:** allow storing embeddables as objects ([#927](https://github.com/mikro-orm/mikro-orm/issues/927)) ([ba881e6](https://github.com/mikro-orm/mikro-orm/commit/ba881e6257dd5d72bb10ca402b0322f7dbbda69c)), closes [#906](https://github.com/mikro-orm/mikro-orm/issues/906)





# [4.1.0](https://github.com/mikro-orm/mikro-orm/compare/v4.0.7...v4.1.0) (2020-10-12)


### Bug Fixes

* **mongo:** filter by serialized PK inside group condition ([a492a64](https://github.com/mikro-orm/mikro-orm/commit/a492a64e08fc4fac5b447d0928bbc6d7b453e29f)), closes [#908](https://github.com/mikro-orm/mikro-orm/issues/908)
* **typings:** improve inference of the entity type ([67f8015](https://github.com/mikro-orm/mikro-orm/commit/67f80157ae013479b6fc47ae1c08a5cd31a6c32d)), closes [#876](https://github.com/mikro-orm/mikro-orm/issues/876)


### Performance Improvements

* **core:** implement bulk updates in mongo driver ([5f347c1](https://github.com/mikro-orm/mikro-orm/commit/5f347c1de4e5dd6f30305275c86d333611edc27c)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)
* **core:** implement bulk updates in sql drivers ([b005353](https://github.com/mikro-orm/mikro-orm/commit/b00535349368ba18a5e0a5452ae7e2b567e12952)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)
* **core:** optimize entity hydration ([6c56a05](https://github.com/mikro-orm/mikro-orm/commit/6c56a05a86b78fc9c3ebc6ddceb75072289e6b48)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)
* **core:** use bulk inserts in all drivers ([10f2e55](https://github.com/mikro-orm/mikro-orm/commit/10f2e55f6710822c9fca272b01c278ff80065096)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)
* **core:** use dedicated identity maps for each entity ([84667f9](https://github.com/mikro-orm/mikro-orm/commit/84667f9e97323e4b054db2c0c70939ab0ca86c86)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)
* **core:** use faster way to check number of object keys ([82f3ee4](https://github.com/mikro-orm/mikro-orm/commit/82f3ee4d4169def8ce8fe31764171193e8b8b5dc)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)
* **core:** use JIT compiled PK getters/serializers ([0ec99dc](https://github.com/mikro-orm/mikro-orm/commit/0ec99dc75690bf15df3897b4da0fc3b2ab709cdd)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)





## [4.0.7](https://github.com/mikro-orm/mikro-orm/compare/v4.0.6...v4.0.7) (2020-09-24)


### Features

* **core:** add groupBy, having and schema to `CountOptions` ([d3c3858](https://github.com/mikro-orm/mikro-orm/commit/d3c38584c38e11002460a6556405e136aabefa93))


### Performance Improvements

* **core:** use batch inserts in UoW (postgres & mongodb) ([#865](https://github.com/mikro-orm/mikro-orm/issues/865)) ([54ad928](https://github.com/mikro-orm/mikro-orm/commit/54ad928aab44d0c42e3f7b306eef0c07ed65dfc1)), closes [#732](https://github.com/mikro-orm/mikro-orm/issues/732)





## [4.0.6](https://github.com/mikro-orm/mikro-orm/compare/v4.0.5...v4.0.6) (2020-09-22)

**Note:** Version bump only for package @mikro-orm/mongodb





## [4.0.5](https://github.com/mikro-orm/mikro-orm/compare/v4.0.4...v4.0.5) (2020-09-21)

**Note:** Version bump only for package @mikro-orm/mongodb





## [4.0.4](https://github.com/mikro-orm/mikro-orm/compare/v4.0.3...v4.0.4) (2020-09-19)

**Note:** Version bump only for package @mikro-orm/mongodb





## [4.0.3](https://github.com/mikro-orm/mikro-orm/compare/v4.0.2...v4.0.3) (2020-09-15)

**Note:** Version bump only for package @mikro-orm/mongodb





## [4.0.2](https://github.com/mikro-orm/mikro-orm/compare/v4.0.1...v4.0.2) (2020-09-11)

**Note:** Version bump only for package @mikro-orm/mongodb





## [4.0.1](https://github.com/mikro-orm/mikro-orm/compare/v4.0.0...v4.0.1) (2020-09-10)

**Note:** Version bump only for package @mikro-orm/mongodb
