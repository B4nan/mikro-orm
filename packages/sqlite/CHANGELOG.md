# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [4.0.3](https://github.com/mikro-orm/mikro-orm/compare/v4.0.2...v4.0.3) (2020-09-15)


### Performance Improvements

* **core:** create the helper instance early ([f4f90eb](https://github.com/mikro-orm/mikro-orm/commit/f4f90ebe7d540164c2959c554a5245dda712b612))
* **core:** do not use `em.merge()` internally ([6a1a6d6](https://github.com/mikro-orm/mikro-orm/commit/6a1a6d68b65a20b8f1a78bf644844427f3b2dd1a))





## [4.0.2](https://github.com/mikro-orm/mikro-orm/compare/v4.0.1...v4.0.2) (2020-09-11)

**Note:** Version bump only for package @mikro-orm/sqlite





## [4.0.1](https://github.com/mikro-orm/mikro-orm/compare/v4.0.0...v4.0.1) (2020-09-10)

**Note:** Version bump only for package @mikro-orm/sqlite
