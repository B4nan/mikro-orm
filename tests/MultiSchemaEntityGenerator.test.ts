import { pathExists, remove } from 'fs-extra';
import { AbstractSqlDriver, DatabaseTable } from '@mikro-orm/knex';
import { EntityGenerator } from '@mikro-orm/entity-generator';
import { BASE_DIR, initORMMySql, initORMPostgreSql, initORMSqlite, wipeDatabase } from './bootstrap';
import { JavaScriptMetadataProvider, MikroORM } from '@mikro-orm/core';
import { FooBarSchema2, FooBazSchema2 } from './entities-with-schema';
import { SqliteDriver } from '@mikro-orm/sqlite';
import { PostgreSqlDriver } from '@mikro-orm/postgresql';
import { MongoDriver } from '@mikro-orm/mongodb';

describe('EntityGenerator', () => {

  test('generate entities from schema with schema name [mysql]', async () => {
    const orm = await MikroORM.init<AbstractSqlDriver>({
      entities: [FooBarSchema2, FooBazSchema2],
      dbName: 'mikro_orm_test',
      explicitSchemaName: true,
      multipleStatements: true,
      debug: ['info'],
      timezone: 'Z',
      charset: 'utf8mb4',
      port: 3307,
      baseDir: BASE_DIR,
      type: 'mysql',
    });

    await orm.getSchemaGenerator().ensureDatabase();
    const connection = orm.em.getConnection();
    await connection.loadFile(__dirname + '/mysql-multi-schema.sql');
    const generator = orm.getEntityGenerator();
    const dump = await generator.generate({ save: true, baseDir: './temp/entities', schemas: ['mikro_orm_test_multi_1', 'mikro_orm_test_multi_2'] });
    expect(dump).toMatchSnapshot('mysql-entity-dump-with-explicit-schema');
    await expect(pathExists('./temp/entities/FooBarSchema2.ts')).resolves.toBe(true);
    await remove('./temp/entities');

    await orm.close(true);
  });

  test('generate entities from schema with schema name [mariadb]', async () => {
    const orm = await MikroORM.init<AbstractSqlDriver>({
      entities: [FooBarSchema2, FooBazSchema2],
      dbName: 'mikro_orm_test',
      explicitSchemaName: true,
      multipleStatements: true,
      debug: ['info'],
      timezone: 'Z',
      charset: 'utf8mb4',
      port: 3309,
      baseDir: BASE_DIR,
      type: 'mysql',
    });

    await orm.getSchemaGenerator().ensureDatabase();
    const connection = orm.em.getConnection();
    await connection.loadFile(__dirname + '/mysql-multi-schema.sql');
    const generator = orm.getEntityGenerator();
    const dump = await generator.generate({ save: true, baseDir: './temp/entities', schemas: ['mikro_orm_test_multi_1', 'mikro_orm_test_multi_2'] });
    expect(dump).toMatchSnapshot('mariadb-entity-dump-with-explicit-schema');
    await expect(pathExists('./temp/entities/FooBarSchema2.ts')).resolves.toBe(true);
    await remove('./temp/entities');

    await orm.close(true);
  });

  test('does not generate entities from schema with schema name even though requested [sqlite]', async () => {
    const orm = await MikroORM.init<SqliteDriver>({
      entities: [FooBarSchema2, FooBazSchema2],
      dbName: ':memory:',
      baseDir: BASE_DIR,
      driver: SqliteDriver,
      explicitSchemaName: true,
      multipleStatements: true,
      debug: ['info'],
      forceUtcTimezone: true,
      cache: { enabled: true, pretty: true },
    });

    await orm.getSchemaGenerator().ensureDatabase();
    const connection = orm.em.getConnection();
    await connection.loadFile(__dirname + '/sqlite-schema.sql');
    const generator = orm.getEntityGenerator();
    const dump = await generator.generate({ save: true, baseDir: './temp/entities', schemas: ['mikro_orm_test_multi_1', 'mikro_orm_test_multi_2'] });
    expect(dump).toMatchSnapshot('sqlite-entity-dump-with-explicit-schema');
    await expect(pathExists('./temp/entities/Publisher3.ts')).resolves.toBe(true);
    await remove('./temp/entities');

    await orm.close(true);
  });

  test('generate entities from schema with schema name [postgresql]', async () => {
    const orm = await MikroORM.init<PostgreSqlDriver>({
      entities: [FooBarSchema2, FooBazSchema2],
      dbName: 'postgres',
      explicitSchemaName: true,
      multipleStatements: true,
      debug: ['info'],
      timezone: 'Z',
      charset: 'utf8',
      port: 5432,
      baseDir: BASE_DIR,
      type: 'postgresql',
    });

    await orm.getSchemaGenerator().ensureDatabase();
    const connection = orm.em.getConnection();
    await connection.loadFile(__dirname + '/postgres-multi-schema.sql');
    const generator = orm.getEntityGenerator();
    const dump = await generator.generate({ save: true, baseDir: './temp/entities', schemas: ['mikro_orm_test_multi_1', 'mikro_orm_test_multi_2'] });
    expect(dump).toHaveLength(3);
    expect(dump).toMatchSnapshot('postgresql-entity-dump-with-explicit-schema');
    await expect(pathExists('./temp/entities/FooBarSchema2.ts')).resolves.toBe(true);
    await remove('./temp/entities');

    await orm.close(true);
  });

});
