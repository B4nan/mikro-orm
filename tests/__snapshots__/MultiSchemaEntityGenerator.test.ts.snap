// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityGenerator does not generate entities from schema with schema name even though requested [sqlite]: sqlite-entity-dump-with-explicit-schema 1`] = `
Array [
  "import { Entity, Index, ManyToOne, PrimaryKey, Property, Unique } from '@mikro-orm/core';
import { Book3 } from './Book3';

@Entity()
@Index({ name: 'author3_name_favourite_book_id_index', properties: ['name', 'favouriteBook'] })
export class Author3 {

  @PrimaryKey()
  id!: number;

  @Property({ nullable: true })
  createdAt?: Date;

  @Property({ nullable: true })
  updatedAt?: Date;

  @Property()
  name!: string;

  @Unique({ name: 'author3_email_unique' })
  @Property()
  email!: string;

  @Property({ nullable: true })
  age?: number;

  @Property()
  termsAccepted!: number;

  @Property({ nullable: true })
  identities?: string;

  @Property({ columnType: 'date(3)', nullable: true })
  born?: string;

  @Property({ columnType: 'time(3)', nullable: true })
  bornTime?: string;

  @ManyToOne({ entity: () => Book3, onUpdateIntegrity: 'cascade', nullable: true, index: 'author3_favourite_book_idx' })
  favouriteBook?: Book3;

}
",
  "import { Entity, ManyToOne, PrimaryKey, Property } from '@mikro-orm/core';
import { Author3 } from './Author3';
import { Publisher3 } from './Publisher3';

@Entity()
export class Book3 {

  @PrimaryKey()
  id!: number;

  @Property({ nullable: true })
  createdAt?: Date;

  @Property({ nullable: true })
  updatedAt?: Date;

  @Property({ default: '' })
  title!: string;

  @ManyToOne({ entity: () => Author3, onUpdateIntegrity: 'cascade' })
  author!: Author3;

  @ManyToOne({ entity: () => Publisher3, onUpdateIntegrity: 'cascade' })
  publisher!: Publisher3;

}
",
  "import { Entity, ManyToOne, PrimaryKey } from '@mikro-orm/core';
import { Book3 } from './Book3';
import { BookTag3 } from './BookTag3';

@Entity({ collection: 'book3_tags' })
export class Book3Tags {

  @PrimaryKey()
  id!: number;

  @ManyToOne({ entity: () => Book3, onUpdateIntegrity: 'cascade', onDelete: 'cascade' })
  book3!: Book3;

  @ManyToOne({ entity: () => BookTag3, onUpdateIntegrity: 'cascade', onDelete: 'cascade' })
  bookTag3!: BookTag3;

}
",
  "import { Entity, PrimaryKey, Property } from '@mikro-orm/core';

@Entity()
export class BookTag3 {

  @PrimaryKey()
  id!: number;

  @Property()
  name!: string;

  @Property({ defaultRaw: \`current_timestamp\` })
  version!: Date;

}
",
  "import { Entity, PrimaryKey, Property } from '@mikro-orm/core';

@Entity()
export class Publisher3 {

  @PrimaryKey()
  id!: number;

  @Property()
  name!: string;

  @Property()
  type!: string;

}
",
  "import { Entity, ManyToOne, PrimaryKey } from '@mikro-orm/core';
import { Publisher3 } from './Publisher3';
import { Test3 } from './Test3';

@Entity({ collection: 'publisher3_tests' })
export class Publisher3Tests {

  @PrimaryKey()
  id!: number;

  @ManyToOne({ entity: () => Publisher3, onUpdateIntegrity: 'cascade', onDelete: 'cascade' })
  publisher3!: Publisher3;

  @ManyToOne({ entity: () => Test3, onUpdateIntegrity: 'cascade', onDelete: 'cascade' })
  test3!: Test3;

}
",
  "import { Entity, PrimaryKey, Property } from '@mikro-orm/core';

@Entity()
export class Test3 {

  @PrimaryKey()
  id!: number;

  @Property({ nullable: true })
  name?: string;

  @Property()
  version: number = 1;

}
",
]
`;

exports[`EntityGenerator generate entities from schema with schema name [mariadb]: mariadb-entity-dump-with-explicit-schema 1`] = `
Array [
  "import { Entity, OneToOne, PrimaryKey, Property } from '@mikro-orm/core';
import { FooBazSchema2 } from './FooBazSchema2';

@Entity({ schema: 'mikro_orm_test_multi_1' })
export class FooBarSchema2 {

  @PrimaryKey()
  id!: number;

  @Property({ length: 255 })
  name!: string;

  @OneToOne({ entity: () => FooBazSchema2, onUpdateIntegrity: 'cascade', onDelete: 'set null', nullable: true, defaultRaw: \`NULL\` })
  baz?: FooBazSchema2;

  @OneToOne({ entity: () => FooBarSchema2, onUpdateIntegrity: 'cascade', onDelete: 'set null', nullable: true, defaultRaw: \`NULL\` })
  fooBar?: FooBarSchema2;

  @Property({ defaultRaw: \`current_timestamp()\` })
  version!: Date;

  @Property({ length: 65535, nullable: true, defaultRaw: \`NULL\` })
  blob?: Buffer;

  @Property({ columnType: 'text', length: 65535, nullable: true, defaultRaw: \`NULL\` })
  array?: string;

  @Property({ columnType: 'longtext', length: 4294967295, nullable: true, defaultRaw: \`NULL\` })
  object?: unknown;

}
",
  "import { Entity, PrimaryKey, Property } from '@mikro-orm/core';

@Entity({ collection: 'foo_bar_schema2', schema: 'mikro_orm_test_multi_2' })
export class MikroOrmTestMulti2fooBarSchema2 {

  @PrimaryKey()
  id!: number;

  @Property({ length: 255 })
  name!: string;

  @Property({ nullable: true })
  bazId?: number;

  @Property({ nullable: true })
  fooBarId?: number;

  @Property({ defaultRaw: \`current_timestamp()\` })
  version!: Date;

  @Property({ length: 65535, nullable: true, defaultRaw: \`NULL\` })
  blob?: Buffer;

  @Property({ columnType: 'text', length: 65535, nullable: true, defaultRaw: \`NULL\` })
  array?: string;

  @Property({ columnType: 'longtext', length: 4294967295, nullable: true, defaultRaw: \`NULL\` })
  object?: unknown;

}
",
  "import { Entity, PrimaryKey, Property } from '@mikro-orm/core';

@Entity({ schema: 'mikro_orm_test_multi_2' })
export class FooBazSchema2 {

  @PrimaryKey()
  id!: number;

  @Property({ length: 255 })
  name!: string;

  @Property({ length: 3, defaultRaw: \`current_timestamp(3)\` })
  version!: Date;

}
",
]
`;

exports[`EntityGenerator generate entities from schema with schema name [mysql]: mysql-entity-dump-with-explicit-schema 1`] = `
Array [
  "import { Entity, OneToOne, PrimaryKey, Property } from '@mikro-orm/core';
import { FooBazSchema2 } from './FooBazSchema2';

@Entity({ schema: 'mikro_orm_test_multi_1' })
export class FooBarSchema2 {

  @PrimaryKey()
  id!: number;

  @Property({ length: 255 })
  name!: string;

  @OneToOne({ entity: () => FooBazSchema2, onUpdateIntegrity: 'cascade', onDelete: 'set null', nullable: true })
  baz?: FooBazSchema2;

  @OneToOne({ entity: () => FooBarSchema2, onUpdateIntegrity: 'cascade', onDelete: 'set null', nullable: true })
  fooBar?: FooBarSchema2;

  @Property({ defaultRaw: \`CURRENT_TIMESTAMP\` })
  version!: Date;

  @Property({ length: 65535, nullable: true })
  blob?: Buffer;

  @Property({ columnType: 'text', length: 65535, nullable: true })
  array?: string;

  @Property({ columnType: 'json', nullable: true })
  object?: any;

}
",
  "import { Entity, PrimaryKey, Property } from '@mikro-orm/core';

@Entity({ collection: 'foo_bar_schema2', schema: 'mikro_orm_test_multi_2' })
export class MikroOrmTestMulti2fooBarSchema2 {

  @PrimaryKey()
  id!: number;

  @Property({ length: 255 })
  name!: string;

  @Property({ nullable: true })
  bazId?: number;

  @Property({ nullable: true })
  fooBarId?: number;

  @Property({ defaultRaw: \`CURRENT_TIMESTAMP\` })
  version!: Date;

  @Property({ length: 65535, nullable: true })
  blob?: Buffer;

  @Property({ columnType: 'text', length: 65535, nullable: true })
  array?: string;

  @Property({ columnType: 'json', nullable: true })
  object?: any;

}
",
  "import { Entity, PrimaryKey, Property } from '@mikro-orm/core';

@Entity({ schema: 'mikro_orm_test_multi_2' })
export class FooBazSchema2 {

  @PrimaryKey()
  id!: number;

  @Property({ length: 255 })
  name!: string;

  @Property({ length: 3, defaultRaw: \`current_timestamp(3)\` })
  version!: Date;

}
",
]
`;

exports[`EntityGenerator generate entities from schema with schema name [postgresql]: postgresql-entity-dump-with-explicit-schema 1`] = `
Array [
  "import { Entity, PrimaryKey, Property } from '@mikro-orm/core';

@Entity({ schema: 'mikro_orm_test_multi_1' })
export class FooBarSchema2 {

  @PrimaryKey()
  id!: number;

  @Property({ length: 255 })
  name!: string;

  @Property({ nullable: true })
  bazId?: number;

  @Property({ nullable: true })
  fooBarId?: number;

  @Property({ length: 6, defaultRaw: \`CURRENT_TIMESTAMP\` })
  version!: Date;

  @Property({ nullable: true })
  blob?: Buffer;

  @Property({ columnType: 'text', nullable: true })
  array?: string;

  @Property({ columnType: 'json', nullable: true })
  object?: any;

}
",
  "import { Entity, PrimaryKey, Property } from '@mikro-orm/core';

@Entity({ collection: 'foo_bar_schema2', schema: 'mikro_orm_test_multi_2' })
export class MikroOrmTestMulti2fooBarSchema2 {

  @PrimaryKey()
  id!: number;

  @Property({ length: 255 })
  name!: string;

  @Property({ nullable: true })
  bazId?: number;

  @Property({ nullable: true })
  fooBarId?: number;

  @Property({ length: 6, defaultRaw: \`CURRENT_TIMESTAMP\` })
  version!: Date;

  @Property({ nullable: true })
  blob?: Buffer;

  @Property({ columnType: 'text', nullable: true })
  array?: string;

  @Property({ columnType: 'json', nullable: true })
  object?: any;

}
",
  "import { Entity, PrimaryKey, Property } from '@mikro-orm/core';

@Entity({ schema: 'mikro_orm_test_multi_2' })
export class FooBazSchema2 {

  @PrimaryKey()
  id!: number;

  @Property({ length: 255 })
  name!: string;

  @Property({ length: 6, defaultRaw: \`current_timestamp(6)\` })
  version!: Date;

}
",
]
`;
